File: CSharp/1125.cs
using System;
using System.Linq;
using System.Collections.Generic;
class URI 
{
    static void Main(string[] args) 
    {
        int[] gp;
        List<int> resultadoCorrida;
        List<int> pontuacaoPilotos = new List<int>();
        List<int> sistemaDePontuacao = new List<int>();
        int sistemas;
        int[] sistema;
        List<int> print = new List<int>();

        while(true)
        {
            gp =Console.ReadLine().Split().Select(int.Parse).ToArray();
            if(gp[0]==0 &&gp[1]==0)
            {break;}

            List<List<int>> colocacoes = new List<List<int>>();
            for (int i = 0; i < gp[1]; i++)
            {
                colocacoes.Add(new List<int>());
            }

            for(int i=0; i < gp[0];i++)
            {
                resultadoCorrida=Console.ReadLine().Split().Select(int.Parse).ToList();
                for(int j=0;j<gp[1];j++)
                {
                    colocacoes[j].Add(resultadoCorrida[j]);
                }                
            }
            sistemas = int.Parse(Console.ReadLine());
            for(int i=0;i<sistemas;i++)
            {
                sistema =Console.ReadLine().Split().Select(int.Parse).ToArray();
                for(int j=0; j<gp[1];j++)
                {
                    pontuacaoPilotos.Add(0);
                    for(int s=1;s<=sistema[0];s++)
                    {
                        pontuacaoPilotos[j]+=sistema[s]*colocacoes[j].Count(x=>x==s);
                    }
                }
                for(int indice=0; indice<gp[1];indice++)
                {
                    if(pontuacaoPilotos[indice]==pontuacaoPilotos.Max())
                    {
                        print.Add(indice+1);
                    }
                }
                Console.WriteLine(string.Join(" ",print));
                print.Clear();
                pontuacaoPilotos.Clear();
            }

        }
    }
}
        

File: Python/1125.py
while True:
    gp = list(map(int, input().split()))
    if gp[0] == 0 and gp[1] == 0:
        break

    colocacoes = [[] for _ in range(gp[1])]

    for _ in range(gp[0]):
        resultadoCorrida = list(map(int, input().split()))
        for j in range(gp[1]):
            colocacoes[j].append(resultadoCorrida[j])

    sistemas = int(input())
    for _ in range(sistemas):
        sistema = list(map(int, input().split()))
        pontuacaoPilotos = [0] * gp[1]
        
        for j in range(gp[1]):
            for s in range(1, sistema[0] + 1):
                pontuacaoPilotos[j] += sistema[s] * colocacoes[j].count(s)
        
        max_pontuacao = max(pontuacaoPilotos)
        indices_vencedores = []
        for indice in range(gp[1]):
            if pontuacaoPilotos[indice] == max_pontuacao:
                indices_vencedores.append(indice + 1)
        
        print(" ".join(map(str, indices_vencedores)))
